src/book/02-system/01-foundations/03-continuous-integration.mdx

```diff
@@ -39,155 +49,31 @@ This CI process works in combination with the [Git Feature Branch workflow](http
 
-```bash
-# Install Python 3 and pip
-sudo apt-get update
-sudo apt-get install python3.6 python3-pip
+## Workflow Outline
 
-# Install Docker Compose
-sudo -H pip3 install docker-compose
-```
+**Note that the source of truth for what a workflow actually does is the given workflow file. It defines precisely what happens.**
 
-### Install and Configure Buildkite
+When a pull request is created or its branch is modified, checks are run on the code to make sure that it builds and maintains standards of correctness. First, an action checks which parts of the codebase have been modified, which informs which checks should be run. For instance, changes to C++ code require the C++ code to be built, and its formatting checked.
 
-Steps from <https://buildkite.com/docs/agent/v3/ubuntu>.
+Next, a docker image tag is built and pushed to DockerHub for the PR. This is cached from the PR image, if it already exists, and the `main` branch image. The caching means that only what needs to be rebuilt in the image is rebuilt, to the point that if the image hasn't changed, the PR image tag simply acts as an alias for the image it cached from.
 
-```bash
-# Add the signed Buildkite apt repository
-echo "deb https://apt.buildkite.com/buildkite-agent stable main" | sudo tee /etc/apt/sources.list.d/buildkite-agent.list
-sudo apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 32A37959C2FA5C3C99EFBC32A79206696452D198
+Finally, all other checks are run conditionally, based on whether the files which those checks apply to have changed. These checks are run inside the Docker container associated with the PR image.
 
-# Install the Buildkite agent
-sudo apt-get update
-sudo apt-get install -y buildkite-agent
+The results from each of these steps are displayed on the pull request page on GitHub (see below).
 
-# Add the agent token to the configuration file.
-# Replace INSERT-YOUR-AGENT-TOKEN-HERE with the token from Buildkite.
-sudo sed -i "s/xxx/INSERT-YOUR-AGENT-TOKEN-HERE/g" /etc/buildkite-agent/buildkite-agent.cfg
+Each of these steps happens when a pull request is merged into the `main` branch. The results are displayed on the main page of the repository though, rather than the pull request page.
 
-# Start the agent and configure it to run on system startup
-sudo systemctl enable buildkite-agent && sudo systemctl start buildkite-agent
-```
+Additionally, when pull requests are merged into main, docker images are built, tagged, and pushed to be the new standard images we use - for example, the `generic` image used for most of our development. This actions workflow is the primary way that those standard images are updated. They can be updated manually by pushing a local image, but this should only be done in exceptional circumstances.
```

Comment:
We should document how to do this at some point but I think it's out of scope for this page

Reply:
Should we document that? I think it's as simple as `docker tag nubots/nubots:generic && docker push nubots/nubots:generic`, which is what you would expect, right? Is that something we need to cover in our docs?

In saying that, I feel like perhaps I know too much. Maybe I have lost touch with what nubook should have, since I am familiar with these tools now.

Reply:
Personally I didn't know how to do that although I would probably have been able to figure it out by looking though the docker documentation. I think it's a classic case of at what point does it change from documentation to tutorials. Probs fine to leave out tbh